device/alpaka/src/utils/parallel_algorithms.hpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * traccc library, part of the ACTS project (R&D line)
+ *
+ * (c) 2025 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Local include(s).
+#include "utils.hpp"
+
+// Project include(s).
+#include "traccc/utils/memory_resource.hpp"
+
+// Thrust include(s).
+#include <thrust/binary_search.h>
+#include <thrust/copy.h>
+#include <thrust/execution_policy.h>
+#include <thrust/fill.h>
+#include <thrust/scan.h>
+#include <thrust/sort.h>
+#include <thrust/unique.h>
```

Comment:
Should these not be hidden when a SYCL build is performed? I know that right now Thrust is always available in our builds, but that may not remain the case always.

Since in the same way we could technically make oneDPL just always be available as well, and could remove the preprocessor statements around those includes as well. (But in general I don't think that would be a good idea.)